### ISSUE TITLE
Behaviour on /repositories/:repository_id/pull. Timeout error

### ISSUE THREAD

**Issue by cacique-coder** [PlaceOS/rest-api#143]
**Is your feature request related to a problem? Please describe.**
- Every time when are pulling, sometimes we are getting a 408 error. That is because how this endpoint works

It is hard to write an script to automatise deployment 

**Describe the solution you'd like**

- when hitting: /repositories/:repository_id/pull return a 200 indicating "pull processing is running"
- then hitting GET /repositories/:repository_id/pull/status => Returning a json with a field "status" This status could be
  - failed
  - in progress
  - completed

**Describe alternatives you've considered**

A possible Web-socket implementation could work, but it would make hard to write code for deployment

**Additional context**

**Comment by cacique-coder** [comment]
Hi guys

I am not sure if you can help me out here. I am working on a terraform provider. Even when back office is good to deploy small services, when there is the need to duplicate/update multiple drivers/modules/system, back office is painful.

Having a provider in terraform this would be easier. However I need some help from your side.

Related to Zones, Repositories, Drivers, etc.
- Which fields can't be updated? 
- On Drivers: how is the mapping for Roles?  99 is Logic, and how about the rest?

Link to Terraform provider: https://github.com/cacique-coder/terraform-placeos-provider

**Comment by stakach** [comment]
the repository pull issue is resolved with the latest release `1.2107`
In terms of the models you can see the details here: https://github.com/PlaceOS/models/tree/master/src/placeos-models

In terms of the roles
```
      SSH       =  0
      Device    =  1 # TCP
      Service   =  2 # HTTP / HTTPS
      Websocket =  3
      Logic     = 99
```
